frontend/src/pages/device/components/DevicePerformance.vue

```diff
@@ -0,0 +1,179 @@
+                            <div class="text-gray-800 text-xl font-medium whitespace-nowrap">CPU Utilisation</div>
+                        </div>
+                    </template>
+                </SectionTopMenu>
+
+                <CpuChart :resources="resources" :device="device" :loading="resources.length === 0" />
+            </section>
+
+            <section class="ff-chart-section">
+                <SectionTopMenu>
+                    <template #hero>
+                        <div class="flex items-center gap-2">
+                            <ChipIcon class="ff-icon ff-icon-md text-gray-800" />
+                            <div class="text-gray-800 text-xl font-medium whitespace-nowrap">Memory Utilisation</div>
+                        </div>
+                    </template>
+                </SectionTopMenu>
+                <MemoryChart :resources="resources" :device="device" :loading="resources.length === 0" />
+            </section>
+        </template>
+    </template>
+</template>
+
+<script>
+import { ChipIcon } from '@heroicons/vue/outline'
+import SemVer from 'semver'
+import { mapState } from 'vuex'
+
+import deviceApi from '../../../api/devices.js'
+import EmptyState from '../../../components/EmptyState.vue'
+import FfLoading from '../../../components/Loading.vue'
+import SectionTopMenu from '../../../components/SectionTopMenu.vue'
+import FeatureUnavailable from '../../../components/banners/FeatureUnavailable.vue'
+import FeatureUnavailableToTeam from '../../../components/banners/FeatureUnavailableToTeam.vue'
+
+import CpuChart from '../../../components/charts/performance/CpuChart.vue'
+import MemoryChart from '../../../components/charts/performance/MemoryChart.vue'
+import usePermissions from '../../../composables/Permissions.js'
+import featuresMixin from '../../../mixins/Features.js'
+
+let mqtt
+
+export default {
+    name: 'DevicePerformanceView',
+    components: {
+        CpuChart,
+        MemoryChart,
+        FfLoading,
+        EmptyState,
+        SectionTopMenu,
+        ChipIcon,
+        FeatureUnavailable,
+        FeatureUnavailableToTeam
+    },
+    mixins: [featuresMixin],
+    inheritAttrs: false,
+    props: {
+        device: {
+            type: Object,
+            required: true
+        }
+    },
+    setup () {
+        const { hasPermission } = usePermissions()
+
+        return {
+            hasPermission
+        }
+    },
+    data () {
+        return {
+            loading: true,
+            resources: [],
+            keepAliveInterval: null,
+            connection: null,
+            client: null
+        }
+    },
+    computed: {
+        ...mapState('account', ['team']),
+        deviceOnline () {
+            const offline = ['stopped', 'offline', 'error']
+            return !offline.includes(this.device.status)
+        },
+        agentSatisfiesVersion () {
+            return this.device && this.device.agentVersion && SemVer.satisfies(this.device.agentVersion, '>=3.5.1')
```

Comment:
```suggestion
            return this.device && this.device.agentVersion && SemVer.satisfies(this.device.agentVersion, '>=3.5.1', { includePrerelease: true })
```

In testing I could not get a beta flag to work - consider adding `includePrerelease: true`